### PR TITLE
[FARM-437] Detail list item for recommendations

### PR DESCRIPTION
**Pull request by davidfranquet**
[FARM-437]

#### 📲 What

New widget for the list items of the recommendation details. Also added all the icons for the recommendation details

#### 🤔 Why
		
Needed for display the details for a crop recommendation
		
#### 🛠 How
		
Created a new widget using a ListTile and adding a title, subtitle (description + list of colors) and a leading (icon)

#### 👀 See
		
Jul-22-2019 16-26-23

		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        (color) => Container(
+              padding: _Constants.circlePadding,
+              child: CustomPaint(
+                painter: _DrawCircle(color: color),
+              ),
+            ),
+      ).toList();
+}
+
+class _DrawCircle extends CustomPainter {
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: does it have to be done with custom painter, or could we just have square views with rounded corners half the width? be interested to see any performance differences.

**Review by LeeMatthewHiggins — DISMISSED** [review]

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart (hunk 1 above)
Comment (davidfranquet) [reply lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: You're right there's a bit difference in FPS. With the custompainter it runs at 52-55 FPS and with simple container with a boxdecoration runs at 56-60 FPS.

Changed!

Comment (LeeMatthewHiggins) [reply lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: perfect! 👍  good job

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart [hunk 2]
```diff
@@ -151,38 +151,12 @@ class RecommendationDetailListItem extends StatelessWidget {
...
+  List<Widget> _buildCircles(List<Color> colors) => colors.map(
         (color) => Container(
-              padding: _Constants.circlePadding,
-              child: CustomPaint(
-                painter: _DrawCircle(color: color),
-              ),
-            ),
+          height: _Constants.circleSize,
+          width:  _Constants.circleSize,
+          decoration: BoxDecoration(color: color, borderRadius: BorderRadius.all(Radius.circular(30))),
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: magic number

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart [hunk 3]
```diff
@@ -151,38 +151,12 @@ class RecommendationDetailListItem extends StatelessWidget {
...
 
-  List<Container> _buildCircles(List<Color> colors) => colors.map(
+  List<Widget> _buildCircles(List<Color> colors) => colors.map(
         (color) => Container(
-              padding: _Constants.circlePadding,
-              child: CustomPaint(
-                painter: _DrawCircle(color: color),
-              ),
-            ),
+          height: _Constants.circleSize,
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: could this be dynamic? so it fits the height maybe or at least be a param, I might want bigger circles in some places

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: 🔮

Comment (davidfranquet) [reply lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: Done!

File: lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/ui/common/recommendation_detail_listitem/recommendation_detail_listitem.dart]: Done!

**Review by LeeMatthewHiggins — APPROVED** [review]
Looks good 👍

**Review by albertmiro — APPROVED** [review]
Well done! 🚀

**Pull request merged**